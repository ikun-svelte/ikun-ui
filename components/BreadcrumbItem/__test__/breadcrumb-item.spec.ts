import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KBreadcrumbItem from '../src';

let host: HTMLElement;

const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KBreadcrumbItem', () => {
	test('props: cls', async () => {
		const instance = new KBreadcrumbItem({
			target: host,
			props: {
				cls: 'k-breadcrumb-item--test'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-breadcrumb-item--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: href is empty', async () => {
		const instance = new KBreadcrumbItem({
			target: host,
			props: {
				href: ''
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-breadcrumb-item-link')).toBeFalsy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: href is not empty', async () => {
		const instance = new KBreadcrumbItem({
			target: host,
			props: {
				href: 'https://github.com/ikun-svelte/ikun-ui'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-breadcrumb-item-link')).toBeTruthy();
		expect(host.querySelector('a')?.getAttribute('href')).toBe(
			'https://github.com/ikun-svelte/ikun-ui'
		);
		expect(host.innerHTML).matchSnapshot();
	});
});
