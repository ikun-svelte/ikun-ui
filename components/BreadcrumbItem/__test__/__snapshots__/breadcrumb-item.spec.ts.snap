// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KBreadcrumbItem > props: cls 1`] = `"<span class=\\"k-breadcrumb-item k-breadcrumb-item--test\\"><span class=\\"k-breadcrumb-item-content k-breadcrumb-item-content__dark\\" href=\\"\\"></span> <span class=\\"k-breadcrumb-item-separator k-breadcrumb-item-separator__dark\\">/</span></span>"`;

exports[`Test: KBreadcrumbItem > props: href is empty 1`] = `"<span class=\\"k-breadcrumb-item\\"><span class=\\"k-breadcrumb-item-content k-breadcrumb-item-content__dark\\" href=\\"\\"></span> <span class=\\"k-breadcrumb-item-separator k-breadcrumb-item-separator__dark\\">/</span></span>"`;

exports[`Test: KBreadcrumbItem > props: href is not empty 1`] = `"<span class=\\"k-breadcrumb-item\\"><a class=\\"k-breadcrumb-item-content k-breadcrumb-item-content__dark k-breadcrumb-item-link\\" href=\\"https://github.com/ikun-svelte/ikun-ui\\"></a> <span class=\\"k-breadcrumb-item-separator k-breadcrumb-item-separator__dark\\">/</span></span>"`;
